### DISCUSSION TITLE
Text to speech

### DISCUSSION THREAD

**Discussion by henriquejourdan** [NightscoutFoundation/xDrip#2197] [Ideas]
Hi
I use Xdrip text to speech function : 5/5 minutes speaks glycemia value.

When master is down (low battery, no Internet or WIFI connections) after 30 minutes, slaves speaks the last glycemia value, which could be dangerous.
Is it possible to change it to MISSING READING from MASTER instead?

Also is it possible to create an alert when the master has low battery (for slaves)?

Would be also great that xdrip also read/speaks the variation value, for example: 110 rising 7

And last but not least: THANKS for developing this software, it helps me a lot with my 5 years kid.

**Comment by Navid200** [comment]
On the follower, you can enable the following setting:

Settings &#8722;> Alarms and Alerts &#8722;> Glucose Alerts Settings &#8722;> Suppress Alerts if missed readings  

Let me know if you still get an alert when there is no reading.
And if you enable missed reading alert, and it doesn't work when there are no readings, let me know.

There are apps that monitor the battery and send an email or text when it crosses a set trigger level.  You can install such apps on the master and set it to email you, follower.  Unfortunately, this redundant functionality also exists in xDrip but not folly functional.

**Comment by henriquejourdan** [comment]
Thanks for the fast reply.
My english is very bad and I probably didnt explained well.
Settings > speak readings > speak readings on
Every 5 minutes i listen the glycemia value. All ok.
But when the master is off or missing readings after 30 minutes, xdrip slave speaks the last working glycemia value.
For example at 18:00 glycemia was 120 mg, from 18:04 to 19:00 the master is offline: no battery. 
At 18:30 xdrip slaves reads the 18:00 glycemia 120 mg ..... if we are luck could be true and real but .... could be completly wrong, specially in  5 years kids.
My ideia is either dont speak anything until master is ok, or speak "missing reading", whatever is easier.

From 18:05 to 19:00 i have also setup a missing read alarm, but if you heard 120 mg from slave you assume that every thing is ok.

About the battery level , we check it in nightscout, but would be great that xdrip have it, specially at night. > master low battery alert.

Once more thanks a lot.
Luis Proenca




-------- Original message --------From: Navid ***@***.***> Date: 29/06/2022  13:39  (GMT+00:00) To: NightscoutFoundation/xDrip ***@***.***> Cc: henriquejourdan ***@***.***>, Author ***@***.***> Subject: Re: [NightscoutFoundation/xDrip] Text to speech (Discussion #2197) 


On the follower, you can enable the following setting:

Settings −> Alarms and Alerts −> Glucose Alerts Settings −> Suppress Alerts if missed readings

Let me know if you still get an alert when there is no reading.


And if you enable missed reading alert, and it doesn't work when there are no readings, let me know.

There are apps that monitor the battery and send an email or text when it crosses a set trigger level.  You can install such apps on the master and set it to email you, follower.  Unfortunately, this redundant functionality also exists in xDrip but not folly functional.



—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/NightscoutFoundation/xDrip/discussions/2197#discussioncomment-3048162",

"url": "https://github.com/NightscoutFoundation/xDrip/discussions/2197#discussioncomment-3048162",

"name": "View Discussion"

},

"description": "View this Discussion on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Reply by Navid200** [reply]
No, it wasn't your explanation.
I completely overlooked what you were explaining.
Makes perfect sense.

First, I need to make sure there isn't already something in place to deal with this:  @tzachi-dar @jamorham @tolot27 

If there isn't, we should make the speak readings conditional on !missed readings.

**Reply by Navid200** [reply]
There is a missed readings alert.
And you can enable "Speak Alerts" under the Speak Readings settings.

I am testing this right now with speak readings enabled to see what happens.

**Comment by Navid200** [comment]
When the master is down (low battery, no Internet or WIFI connections), from the point of view of the follower, there are no readings.  So, all you need to do is to enable the missed readings alert on the follower.
Then, you will be informed, by the alert, that you are not getting new readings anymore.
